frontend/src/pages/projects/screens/detail/data-connections/ManageDataConnectionModal.tsx

```diff
@@ -53,27 +72,62 @@ const ManageDataConnectionModal: React.FC<ManageDataConnectionModalProps> = ({
       existingData?.data.metadata.name,
     );
 
-    const handleError = (e: Error) => {
-      setErrorMessage(e.message || 'An unknown error occurred');
-      setIsProgress(false);
-    };
+    const promiseActions: Promise<K8sResourceCommon>[] = [];
+    const secretName = assembledSecret.metadata.name;
 
     if (existingData) {
-      replaceSecret(assembledSecret)
-        .then(() => onBeforeClose(true))
-        .catch(handleError);
+      if (AWSDataChanged) {
+        promiseActions.push(replaceSecret(assembledSecret));
+      }
+
+      const notebooksToDisconnect = allAvailableNotebooks.filter((notebook) =>
+        removedConnections.includes(notebook.metadata.name),
+      );
+      notebooksToDisconnect.forEach((notebook) => {
```

Comment:
Nit. `promiseActions.push(...notebooksToDisconnect.map(...));`

Reply:
FWIW, alternative is to use `.reduce` and the initial data is the array.

`notebooksToDisconnect.reduce((promises, notebook) => ([...promises, some_logic], promiseActions)`